spring-integration-core/src/main/java/org/springframework/integration/support/management/graph/IntegrationGraphServer.java

```diff
@@ -0,0 +1,185 @@
+				.getBeansOfType(MessageProducerSupport.class);
+		Map<String, IntegrationConsumer> consumers = this.applicationContext.getBeansOfType(IntegrationConsumer.class);
+		Collection<IntegrationNode> nodes = new ArrayList<IntegrationNode>();
+		Collection<LinkNode> links = new ArrayList<LinkNode>();
+		Map<String, MessageChannelNode> channelNodes = new HashMap<String, MessageChannelNode>();
+		for (Entry<String, MessageChannel> entry : channels.entrySet()) {
+			MessageChannel channel = entry.getValue();
+			MessageChannelNode channelNode = this.nodeFactory.channelNode(entry.getKey(), channel);
+			String beanName = entry.getKey();
+			nodes.add(channelNode);
+			channelNodes.put(beanName, channelNode);
+		}
+		for (Entry<String, SourcePollingChannelAdapter> entry : spcas.entrySet()) {
+			SourcePollingChannelAdapter adapter = entry.getValue();
+			MessageSourceNode sourceNode = this.nodeFactory.sourceNode(entry.getKey(), adapter);
+			nodes.add(sourceNode);
+			MessageChannelNode channelNode = channelNodes.get(sourceNode.getOutput());
+			if (channelNode != null) {
+				links.add(new LinkNode(sourceNode.getNodeId(), channelNode.getNodeId()));
+			}
+		}
+		for (Entry<String, MessagingGatewaySupport> entry : gateways.entrySet()) {
+			MessagingGatewaySupport gateway = entry.getValue();
+			MessageGatewayNode gatewayNode = this.nodeFactory.gatewayNode(entry.getKey(), gateway);
+			nodes.add(gatewayNode);
+			MessageChannelNode channelInfo = channelNodes.get(gatewayNode.getOutput());
+			if (channelInfo != null) {
+				links.add(new LinkNode(gatewayNode.getNodeId(), channelInfo.getNodeId()));
+			}
+		}
+		for (Entry<String, MessageProducerSupport> entry : producers.entrySet()) {
+			MessageProducerSupport producer = entry.getValue();
+			MessageProducerNode producerNode = this.nodeFactory.producerNode(entry.getKey(), producer);
+			nodes.add(producerNode);
+			MessageChannelNode channelNode = channelNodes.get(producerNode.getOutput());
+			if (channelNode != null) {
+				links.add(new LinkNode(producerNode.getNodeId(), channelNode.getNodeId()));
+			}
+		}
+		for (Entry<String, IntegrationConsumer> entry : consumers.entrySet()) {
+			IntegrationConsumer consumer = entry.getValue();
+			MessageHandlerNode handlerNode = this.nodeFactory.handlerNode(entry.getKey(), consumer);
+			nodes.add(handlerNode);
+			MessageChannelNode channelNode = channelNodes.get(handlerNode.getInput());
+			if (channelNode != null) {
+				links.add(new LinkNode(channelNode.getNodeId(), handlerNode.getNodeId()));
+			}
+			if (handlerNode.getOutput() != null) {
+				channelNode = channelNodes.get(handlerNode.getOutput());
+				if (channelNode != null) {
+					links.add(new LinkNode(handlerNode.getNodeId(), channelNode.getNodeId()));
+				}
+			}
+		}
+		this.graph = new Graph(nodes, links);
+	}
+
+	/**
+	 * Rebuild the graph if new components have been added.
+	 */
+	public void rebuild() {
+		buildGraph();
```

Comment:
Why do we need two of these `build` & `rebuild` method?
Why just not have that one `buildGraph()` and make it return the `Graph`?
Or... Since we have a `getGraph()` just `build` wouldn't be good. 
And the `rebuildGraph()` with direct return makes more sense and makes the end-user to think twice before using it.
So, for normal last snapshot state the `getGraph()` is good. In case of we would like to have a fresh state, please, call `rebuildGraph()`.
WDYT?